drivers/ethernet/eth_enc424j600.c

```diff
@@ -484,7 +484,11 @@ static void enc424j600_rx_thread(struct enc424j600_runtime *context)
 			}
 		} else {
 			LOG_ERR("Unknown Interrupt, EIR: 0x%04x", eir);
-			continue;
+			/*
+			 * Terminate interrupt handling thread because
+			 * the integrity of the driver is no longer given.
+			 */
+			k_oops();
```

Comment:
I do not accept this, the linux kernel driver ignores any unknown interrupts, no other zephyr ethernet driver does this. This k_oops and the continue needs to go entirely.

Reply:
I am not a big fan of `k_oops()`, the device stop running after that which might not be desirable in embedded device.
Is it so that getting into this branch should not be possible after the changes in second commit?

Reply:
> Is it so that getting into this branch should not be possible after the changes in second commit?

It should never be the case. The second commit fixes a bug that was found thanks to this brunch. Faulty driver should not just keep running. AFAIK we do not have any possibilities to tell the stack or application about an error from driver thread. I might be exaggerating, I could probably live with hiding it behind LOG_LEVEL_DBG, and it would also fit the style of the driver. But logging is for developer, in the field such an error would remains undetected.

```diff c
diff --git a/drivers/ethernet/eth_enc424j600.c b/drivers/ethernet/eth_enc424j600.c
index 417b66ae07..f86718f319 100644
--- a/drivers/ethernet/eth_enc424j600.c
+++ b/drivers/ethernet/eth_enc424j600.c
@@ -484,7 +484,9 @@ static void enc424j600_rx_thread(struct enc424j600_runtime *context)
                        }
                } else {
                        LOG_ERR("Unknown Interrupt, EIR: 0x%04x", eir);
-                       continue;
+                       if (CONFIG_ETHERNET_LOG_LEVEL == LOG_LEVEL_DBG) {
+                               k_oops();
+                       }
                }
 
                enc424j600_set_sfru(context->dev, ENC424J600_SFR3_EIEL,
```

Reply:
The issue that this bug fixed wouldn't have been an issue at all had the continue/k_oops not been there, the driver would have ignore the first spurious interrupt and then continued to process the second valid interrupt and functioned properly. Having just a log output with no continue/k_oops is fine with me.

Reply:
> I could probably live with hiding it behind LOG_LEVEL_DBG

Your proposal looks like a good tradeoff :+1: 

About the error happening in the field, that is certainly possible, but in that case perhaps it is better to ignore/skip the issue than to halt.

Reply:
> The issue that this bug fixed wouldn't have been an issue at all had the continue/k_oops not been there, the driver would have ignore the first spurious interrupt and then continued to process the second valid interrupt and functioned properly. Having just a log output with no continue/k_oops is fine with me.

So you mean that there is no other bug except the one suspected in line 487? So is it okay for your products that there is an interrupt in the Ethernet driver firing for no reason?

Reply:
Drivers and applications are designed to cope with simplistic errors, e.g. a false interrupt, which could be caused by many things including a error in the silicon (of which Microchip note there is a bug in the enc424j600 silicon which causes a spurious interrupt), an error in the MCU silicon or firmware or even from an external source e.g. by a motor. You show me where in the linux kernel (which is used in thousands of embedded devices) whereby a network or modem driver will kill the system if it receives an unexpected interrupt or command: